src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -581,10 +586,73 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
     auto config = selectedPd->getConfig();
     config.outConfs.front().setMemDesc(mem_desc);
-    //bypass any checks, we enforce the child descriptor
+    // bypass any checks, we enforce the child descriptor
     selectedPd->setConfig(config);
 }
 
+void MemoryInput::selectOptimalPrimitiveDescriptor() {
+    if (body) {
+        // for the input configution, just always use the parent configuration
```

Comment:
Actually, I have  an interface `haveSubgraph()`, and replaced them to `haveSubgraph()`

Reply:
This is not really resolved yet.
Instead of always performing this check, it seems we better to introduce some State.
`Initial` state, when we have to infer Init graph and `Execution` state, where we do not care about Init graph anymore.
@maxnick What do you think?

Reply:
It will eventually depend on the memory sharing pattern. To my understanding, since the internal graph is executed pretty rarely (only when the state is reset), we should substitute the memory from the output edge of the MemoryInput node to the internal subgraph, so that the data are written to the MemoryInput dst memory by the internal subgraph. In that case, the memory pointer check routine may be removed, and the only difference with the case without subgraph will be `subGraph.Infer();` call, which still requires the `if(haveSubgaph())` check, but in that case it will be under `if (processInitGraph)`, which is always performed but often has `false` value as we usually don't need to process the init subgraph every inference.